waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/PrimaryMetaStore.java

```diff
@@ -53,17 +45,18 @@ public FederationType getFederationType() {
     return FederationType.PRIMARY;
   }
 
-  @Size(min = 0, max = 0)
   @NotNull
   @Override
   public String getDatabasePrefix() {
-    // primary is always empty
-    return EMPTY_PREFIX;
+    String prefix = super.getDatabasePrefix();
+    if (prefix == null) {
+      prefix = "";
```

Comment:
maybe we keep using `EMPTY_PREFIX` just so it is clear that it is a valid value

Reply:
Done :)